main.rb

```diff
@@ -0,0 +1,171 @@
+             '4:  -  Create a book', '5:  -  Create a rental', '6:  -  List all rentals for a given person id']
+  puts "\n" * 3
+
+  puts '=' * 43
+  puts "|#{' ' * 41}|"
+  puts welcome_message[0].upcase
+  puts "|#{' ' * 41}|"
+  puts '=' * 43
+  puts "\n" * 3
+
+  puts '-' * 49
+  puts welcome_message[1]
+  puts '-' * 49
+  puts "\n" * 2
+
+  puts options
+end
+
+class Storage
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def run
+    choice = show_options
+    choice = show_options while choice < 1 || choice > 7
+
+    case choice
+    when 1
+      handle_book_list
+    when 2
+      handle_person_list
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      handle_rentals_list
+    when 7
+      puts 'Thank you for using this app!'
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def show_options
+    say_welcome
+    gets.chomp.to_i
+  end
+
+  def handle_book_list
+    @books.each { |book| puts "Title: '#{book.title}', Author: #{book.author}" }
+    run
+  end
+
+  def handle_person_list
+    @persons.each do |person|
+      puts "[#{person.class.name.split('::').last}] Name: '#{person.name}', ID: #{person.id} , Age: #{person.age}"
+    end
+    run
+  end
+
+  def handle_rentals_list
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    puts 'Rentals:'
+    selected_rental = @rentals.select { |rental| rental.person.id == id }
+    selected_rental.each do |s_rental|
+      puts "Date: '#{s_rental.date}', Book: #{s_rental.book.title} , by: #{s_rental.book.author}"
+    end
+    run
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @persons.push(Teacher.new(age: age, specialization: specialization, name: name))
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp != 'n'
+    puts 'create'
+    @persons.push(Student.new(age: age, classroom: 'learn to code', name: name, parent_permission: parent_permission))
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = 0
+
+    person_type = gets.chomp while person_type != '2' && person_type != '1'
+
+    case person_type
+    when '1'
+      create_student
+
+    when '2'
+      create_teacher
+    end
+    puts 'Person created successfully'
+    puts
+    run
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    @books.push(Book.new(title: title, author: author))
+    puts 'Book created successfully'
+    puts
```

Comment:
**Changes Required**
- I can tell you have put some effort on implementing the basic UI. However;
- [ ] The create_book, create_person throw an error when I try to create either of the objects. Let us try and fix this behavior.

Reply:
- [ ] Now the app can create a user and a book, but there is a problem when you try to rental a book, please fix it. 🐶 
![image](https://user-images.githubusercontent.com/20563252/137562494-69737436-deb0-4baf-a14d-a39f4d4c13bf.png)